### PR TITLE
Added support for StallFreeINTEL

### PR DESCRIPTION
**Pull request by bwlodarcz**
Commit also adds new extension capability FPGAClusterAttributesV2INTEL. 
Specification:
https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/INTEL/SPV_INTEL_fpga_cluster_attributes.asciidoc

### CODE REVIEW

**Comment by bwlodarcz** [comment]
Feature is now complete but first SPIRV-Headers must merge PR with update.

File: lib/SPIRV/SPIRVWriter.cpp [hunk 1]
```diff
@@ -998,6 +998,15 @@ void LLVMToSPIRVBase::transFPGAFunctionMetadata(SPIRVFunction *BF,
         BF->addDecorate(new SPIRVDecorateStallEnableINTEL(BF));
     }
   }
+  if (MDNode *StallFree = F->getMetadata(kSPIR2MD::StallFree)) {
+    if (BM->isAllowedToUseExtension(
+            ExtensionID::SPV_INTEL_fpga_cluster_attributes)) {
+      if (getMDOperandAsInt(StallFree, 0)) {
+        BM->addCapability(spv::CapabilityFPGAClusterAttributesV2INTEL);
```
Comment (vmaksimo) [inline lib/SPIRV/SPIRVWriter.cpp:1005]: Not sure if need to add capability explicitly. Will the test pass if you remove this line?

**Review by vmaksimo — COMMENTED** [review]
Also please fix clang-format.

Here is a Headers PR for other reviewers reference: https://github.com/KhronosGroup/SPIRV-Headers/pull/393

File: lib/SPIRV/SPIRVWriter.cpp (hunk 1 above)
Comment (bwlodarcz) [reply lib/SPIRV/SPIRVWriter.cpp]: Right. It's not needed.

**Review by MrSidims — APPROVED** [review]

**Pull request merged**